crates/ruff_python_formatter/src/expression/expr_named_expr.rs

```diff
@@ -16,16 +19,42 @@ impl FormatNodeRule<ExprNamedExpr> for FormatExprNamedExpr {
             value,
             range: _,
         } = item;
-        write!(
-            f,
-            [
-                target.format(),
-                space(),
-                text(":="),
-                space(),
-                value.format(),
-            ]
-        )
+        let comments = f.context().comments().clone();
+
+        // This context, a dangling comment is an end-of-line comment on the same line as the `:=`.
+        let dangling = comments.dangling_comments(item);
+
+        if dangling.is_empty() {
+            write!(
+                f,
+                [
+                    group(&format_args!(target.format(), soft_line_break_or_space())),
+                    text(":="),
+                    space(),
+                    value.format(),
+                ]
+            )
+        } else {
+            write!(
+                f,
+                [
+                    group(&format_args!(target.format(), soft_line_break_or_space())),
+                    text(":="),
+                    dangling_comments(dangling),
+                    hard_line_break(),
+                    value.format(),
+                ]
+            )
+        }
```

Comment:
Nit: Can we move out the common logic. It makes it easier to reason about what changes based on whether there are comments present or not.
```suggestion

		write!(f, [
			group(&format_args!(target.format(), soft_line_break_or_space())),
          	text(":="),
        ])?;
                
        if dangling.is_empty() {
        	space().fmt(f)?;
        } else {
            write!(
                f,
                [
                    dangling_comments(dangling),
                    hard_line_break(),
                ]
            )?;
        }
        
        value.format().fmt(f)
```